### PR TITLE
Add Member.publish_member_name_consent

### PR DESCRIPTION
**Pull request by reginafcompton**
This PR adds a new field to the Member model: `publish_member_name_consent`. The new field will be used to help AHS admin keep track of which members have given permission to publish their name for being a "circle" member.

See child PR for making use of the new field in the UI: https://github.com/americanhandelsociety/americanhandelsociety-members/pull/182.

@ecedmondson - I am still waiting on final feedback from the AHS president and newsletter editor on this and its child PR. However, you are welcome to review now, if you like! (I do not anticipate much more will change.)

### CODE REVIEW

File: americanhandelsociety_app/models.py [hunk 1]
```diff
@@ -110,6 +109,11 @@ def max_length(cls):
         auto_now=True,
         null=True,
     )
+    publish_member_name_consent = models.CharField(
```
Comment (reginafcompton) [inline americanhandelsociety_app/models.py:112]: offff. I struggled with this column name.

File: americanhandelsociety_app/migrations/0009_member_publish_member_name_consent.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    operations = [
+        migrations.AddField(
+            model_name="member",
+            name="publish_member_name_consent",
+            field=models.CharField(
+                choices=[
+                    ("NO", "No"),
+                    ("YES", "Yes"),
+                    ("ANONYMOUS", "Yes, but display member name as 'Anonymous'"),
```
Comment (reginafcompton) [inline americanhandelsociety_app/migrations/0009_member_publish_member_name_consent.py:19]: The AHS president asked for a nuanced distinction between "no" and "show-me-as-anonymous." So, here we are.

**Comment by reginafcompton** [comment]
Okay! 

@ecedmondson - I _did_ make another adjustment per feedback from the AHS board. And now: this is ready for "official" official review. Like usual, there is no rush to get this reviewed and deployed. Thanks!

**Comment by ecedmondson** [comment]
>  I did make another adjustment per feedback from the AHS board. And now: this is ready for "official" official review. Like usual, there is no rush to get this reviewed and deployed. Thanks!

Good to know! I have been ordering some personal matters this past week, so I didn't see this until now. It is late, so I probably will not get to this PR and the child PR until sometime this week. Thanks for your patience!

c.f. @reginafcompton

File: americanhandelsociety_app/models_utils.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from django.db import models
+
+
+class BaseTextChoices(models.TextChoices):
+    @classmethod
+    def max_length(cls):
+        return max(*(len(choice.value) for choice in cls))
```
Comment (ecedmondson) [inline americanhandelsociety_app/models_utils.py:7]: I know that this just being copied from the other, existing code, but it is in a new file which draws my attention to it anew. I think `max()` can take a generator. Non-blocking, of course.

File: americanhandelsociety_app/models.py (hunk 1 above)
Comment (ecedmondson) [reply americanhandelsociety_app/models.py:112]: I think this name is fine! Do you want to brainstorm other ideas for it or are you settled on this one?

**Review by ecedmondson — APPROVED** [review]
Code looks good. I will check out the child PR. Unfortunately, I have merge-conflicted you (I did not realize I had merge conflicted you) and also blocked us with a failed test in the very same merge. I have an issue created and will fix the test.

**Comment by reginafcompton** [comment]
I will un-conflict the conflicts before end of week.

File: americanhandelsociety_app/models_utils.py (hunk 3 above)
Comment (reginafcompton) [reply americanhandelsociety_app/models_utils.py:7]: oh, so: I do not need the `*`, since it can just make sense of `(...)`.

**Pull request merged**